controllers/serveController.js

```diff
@@ -82,53 +92,112 @@ function getAssetByClaimId (fullClaimId, name) {
+      })
+      // 3. get the asset by this claim id and name
       .then(claimId => {
-        logger.debug('claim id =', claimId);
-        resolve(getAssetByClaimId(claimId, name));
+        logger.debug('asset claim id = ', claimId);
+        resolve(getAssetByLongClaimId(claimId, claimName));
       })
       .catch(error => {
         reject(error);
       });
     });
   },
-  getAssetByClaimId (fullClaimId, name) {
-    return getAssetByClaimId(fullClaimId, name);
-  },
-  getAssetByName (name) {
-    logger.debug('...getting asset by claim name...');
+  getChannelContents (channelName, channelId) {
     return new Promise((resolve, reject) => {
-      // 1. get a list of the free public claims
-      getTopFreeClaim(name)
-      // 2. check locally for the top claim
-      .then(topFreeClaim => {
-        // if no claims were found, return null
-        if (!topFreeClaim) {
-          return resolve(null);
+      let longChannelId;
+      let shortChannelId;
+      // 1. get the long channel Id
+      getLongChannelId(channelName, channelId)
+      // 2. get all claims for that channel
+      .then(result => {
+        longChannelId = result;
+        return getShortChannelIdFromLongChannelId(channelName, longChannelId);
+      })
+      // 3. get all Claim records for this channel
+      .then(result => {
+        shortChannelId = result;
+        return getAllChannelClaims(longChannelId);
+      })
+      // 4. add extra data not available from Claim table
+      .then(allChannelClaims => {
+        if (allChannelClaims) {
+          allChannelClaims.forEach(element => {
+            element['channelName'] = channelName;
+            element['longChannelId'] = longChannelId;
+            element['shortChannelId'] = shortChannelId;
+            element['fileExtension'] = element.contentType.substring(element.contentType.lastIndexOf('/') + 1);
+          });
         }
-        // parse the result
-        const claimId = topFreeClaim.claimId;
-        logger.debug('top free claim id =', claimId);
-        // get the asset
-        resolve(getAssetByClaimId(claimId, name));
+        return resolve(allChannelClaims);
       })
       .catch(error => {
         reject(error);
       });
     });
   },
+  serveOrShowAsset (fileInfo, extension, method, headers, originalUrl, ip, res) {
+    // add file extension to the file info
+    if (extension === '.gifv') {
+      fileInfo['fileExt'] = '.gifv';
+    } else {
+      fileInfo['fileExt'] = fileInfo.fileName.substring(fileInfo.fileName.lastIndexOf('.'));
+    }
+    // serve or show
+    switch (method) {
+      case SERVE:
+        serveFile(fileInfo, res);
+        sendGoogleAnalytics(method, headers, ip, originalUrl);
+        postToStats('serve', originalUrl, ip, fileInfo.name, fileInfo.claimId, 'success');
+        return fileInfo;
+      case SHOWLITE:
+        showFileLite(fileInfo, res);
+        postToStats('show', originalUrl, ip, fileInfo.name, fileInfo.claimId, 'success');
```

Comment:
This is shared across all 3 cases and could be pulled above/out.

Reply:
https://github.com/lbryio/spee.ch/commit/c901ced94c1d5e2212bc3e6bda25939a6269f028